core/src/main/java/com/orientechnologies/orient/core/engine/OMemoryAndLocalPaginatedEnginesInitializer.java

```diff
@@ -73,6 +84,75 @@ private void configureDefaults() {
       configureDefaultWalRestoreBatchSize();
   }
 
+  private void configureDefaultSoftRefsInSQL() {
+    if (OGlobalConfiguration.QUERY_USE_SOFT_REFENCES_IN_RESULT_SET.getValue() == null) {
+      OLogManager.instance()
+          .infoNoDb(this, "Configuration of usage of soft references inside of containers of results of SQL execution");
+
+      RuntimeMXBean runtimeMXBean = ManagementFactory.getRuntimeMXBean();
+      final List<String> inputArgs = runtimeMXBean.getInputArguments();
+      String xms = null;
+      String xmx = null;
+
+      for (String inputArg : inputArgs) {
+        if (inputArg.matches("-Xms\\d+[gGmMkK]?")) {
+          xms = inputArg;
+        } else if (inputArg.matches("-Xmx\\d+[gGmMkK]?")) {
+          xmx = inputArg;
+        }
+      }
+
+      if (xmx == null || xms == null) {
+        OLogManager.instance().infoNoDb(this, "Initial or maximum values of heap memory usage are NOT set, containers of "
+            + "results of SQL executors will NOT use soft references by default");
+        OGlobalConfiguration.QUERY_USE_SOFT_REFENCES_IN_RESULT_SET.setValue(false);
+        return;
+      }
+
+      final long xmsBytes = extractMemoryLimitInBytes(xms);
+      final long xmxBytes = extractMemoryLimitInBytes(xmx);
+
+      if (xmsBytes == xmxBytes) {
+        OLogManager.instance().infoNoDb(this, "Initial and maximum values of heap memory usage are equal, containers of "
+            + "results of SQL executors will use soft references by default");
+        OGlobalConfiguration.QUERY_USE_SOFT_REFENCES_IN_RESULT_SET.setValue(true);
+      } else {
+        OLogManager.instance().infoNoDb(this, "Initial and maximum values of heap memory usage are NOT equal, containers of "
+            + "results of SQL executors will NOT use soft references by default");
+        OGlobalConfiguration.QUERY_USE_SOFT_REFENCES_IN_RESULT_SET.setValue(false);
+      }
+    }
+  }
+
+  private long extractMemoryLimitInBytes(String limit) {
+    final Pattern pattern = Pattern.compile("((-Xms)|(-Xmx))(\\d+)([gGmMkK]?)");
+    final Matcher matcher = pattern.matcher(limit);
+    if (!matcher.find()) {
+      throw new IllegalArgumentException("Invalid value of memory limit was provided '" + limit + "'");
```

Comment:
why...? There is really no need to throw an exception if you don't explicitly define the memory limit...

Reply:
ah, ok, it's checked above...